tests/eve_tests/unit_tests/test_utils.py

```diff
@@ -369,3 +369,21 @@ def test_xenumerate():
     from gt4py.eve.utils import xenumerate
 
     assert list(xenumerate(string.ascii_letters[:3])) == [(0, "a"), (1, "b"), (2, "c")]
+
+
+def test_lru_cache_key_id_called_once():
+    from gt4py.eve.utils import lru_cache
+
+    call_count = 0
+
+    def func(x):
+        nonlocal call_count
+        call_count += 1
+        return x
+
+    cached = lru_cache(func, key=id)
+
+    obj = object()
+    assert cached(obj) is obj
```

Comment:
You could test the same using the `cache_parameters()` instrumentation function. Please check also the `__wrapped__` attribute is set properly.

Reply:
Testing both now.